plugins/hls-retrie-plugin/src/Ide/Plugin/Retrie.hs

```diff
@@ -333,33 +324,32 @@ provider state plId (CodeActionParams _ _ (TextDocumentIdentifier uri) range ca)
 getLocationUri :: Location -> Uri
 getLocationUri Location{_uri} = _uri
 
+getLocationRange :: Location -> Range
 getLocationRange Location{_range} = _range
 
-getBinds :: NormalizedFilePath -> ExceptT PluginError Action (ModSummary, [HsBindLR GhcRn GhcRn], PositionMapping, [LRuleDecls GhcRn], [TyClGroup GhcRn])
+getBinds :: NormalizedFilePath -> ExceptT PluginError Action
+  ( ModSummary
+  , [HsBindLR GhcRn GhcRn]
+  , PositionMapping
+  , [LRuleDecls GhcRn]
+  , [TyClGroup GhcRn]
+  )
 getBinds nfp = do
   (tm, posMapping) <- useWithStaleE TypeCheck nfp
   -- we use the typechecked source instead of the parsed source
   -- to be able to extract module names from the Ids,
   -- so that we can include adding the required imports in the retrie command
   let rn = tmrRenamed tm
-      ( HsGroup
-          { hs_valds =
-              XValBindsLR
-                (GHC.NValBinds binds _sigs :: GHC.NHsValBindsLR GhcRn),
-            hs_ruleds,
-            hs_tyclds
-          },
-        _,
-        _,
-        _
-        ) = rn
-
-      topLevelBinds =
-        [ decl
-          | (_, bagBinds) <- binds,
-            L _ decl <- bagToList bagBinds
-        ]
-  return (tmrModSummary tm, topLevelBinds, posMapping, hs_ruleds, hs_tyclds)
+  case rn of
+    (HsGroup{hs_valds, hs_ruleds, hs_tyclds}, _, _, _) -> do
+      topLevelBinds <- case hs_valds of
+        ValBinds{} -> throwError $ PluginInternalError "getBinds: ValBinds not supported"
```

Comment:
The above was giving "incomplete pattern match: constructors not matched ValBinds ..." warning.
Throwing error seems far from ideal but I still consider it step forward (better visibility into where the error is coming from).